piggypal/src/Test/ModelTests/ModelUnitTests.py

```diff
@@ -0,0 +1,160 @@
+import os
+import cherrypy
+import requests
+from cherrypy.test import helper
+from contextlib import contextmanager
+from main.Models.Paypal_api_stub import Paypal_cred_listener as pcl, Paypal_stub as ps
+from utils import SRC_DIR
+
+conf = os.path.join(os.path.join(os.path.join(SRC_DIR, "main"), "Configs"), "piggypal.conf")
+
+
+@contextmanager
+def run_server():
+    """
+    Starts cherrypy engine to run tests in proper environment
+    """
+    cherrypy.config.update({'global': {'server.socket_host': "127.0.0.1", 'server.socket_port': 4710}})
+    cherrypy.tree.mount(ps(), '/piggypal', conf)
+    cherrypy.tree.mount(pcl(), '/piggypal-listens', conf)
+    cherrypy.engine.start()
+    cherrypy.engine.wait(cherrypy.engine.states.STARTED)
+    yield
+    cherrypy.engine.exit()
+
+
+class ModelUnitTests(helper.CPWebCase):
+    """
+    Contains Unit Tests for Model
+    """
+    def test_credential_listener_get(self):
+        """
+        Checks if GET saves Paypal Authorization Code
+        """
+        listener = pcl()
+        teststring = "teststring"
+        listener.GET(code=teststring, scope="")
+        self.assertEqual(listener._auth_code, teststring)
+
+    def test_credential_listener_delete(self):
+        """
+        Checks is DELETE returns _auth_code and removes _auth_code object
+
+        Since _auth_code only exists once per session the object should be deleted after getting it.
+        """
+        listener = pcl()
+        teststring = "teststring"
+        listener.GET(code=teststring, scope="")
+        auth_code = listener._auth_code
+        self.assertEqual(auth_code, teststring)
+        auth_code = listener.DELETE()
+        self.assertEqual(auth_code, teststring)
+        try:
+            auth_code = listener._auth_code
+        except AttributeError:
+            pass
+        else:
+            self.assertNotEqual(auth_code, listener._auth_code)
+
+    def test_token_saver(self):
+        """
+        Asserts proper token saving
+        """
+        assertion_token = "TokenTest"
+        stub = ps()
+        stub.tokenSaver(token=assertion_token)
+        self.assertEqual(stub._token, assertion_token)
+
+
+class ModelSandboxUnitTest(helper.CPWebCase):
+    """
+    Contains Unit Tests for Model which can currently be only run under sandbox conditions due to Paypal Partner problems
```

Comment:
Change to: 
"""
Contains Unit Tests for Model

! This Test suite only targets the Paypal Sandbox Environment.
"""